coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py

```diff
@@ -1,10 +1,53 @@
-import os
-#import json
+import json
 import requests
+import boto3
+from dataclasses import dataclass
 
-def lambda_handler(event, context):
 
-    if not 'keys' in event:
+@dataclass
+class CoioteCredentials:
+    url: str
+    username: str
+    password: str
+
+
+def getCoioteSecrets():
+    secretName = "coioteDMrest"
+
+    session = boto3.session.Session()
+    client = session.client('secretsmanager')
+
+    try:
+        getSecretValueResponse = client.get_secret_value(
+            SecretId=secretName
+        )
+    except Exception:
+        raise Exception('Coiote DM credentials not found in Secrets Manager')
+
+    if 'SecretString' not in getSecretValueResponse:
+        raise Exception('Coiote DM credentials are not set up correctly')
+
+    secretsMap = json.JSONDecoder().decode(getSecretValueResponse['SecretString'])
+
+    usernameKey = 'username'
+    passwordKey = 'password'
+    urlKey = 'url'
+
+    expectedKeysSet = {usernameKey, passwordKey, urlKey}
+    fetchedKeysSet = set(secretsMap.keys())
+
+    if expectedKeysSet.issubset(fetchedKeysSet):
+        return CoioteCredentials(
+            url=secretsMap['url'],
+            username=secretsMap['username'],
+            password=secretsMap['password']
```

Comment:
You can use the key strings created earlier

Reply:
That was a point of creating them, I just forgot to change it. Thanks.